src/utils/units.hpp

```diff
@@ -0,0 +1,100 @@
+//========================================================================================
+// (C) (or copyright) 2024. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+#ifndef UTILS_UNITS_HPP_
+#define UTILS_UNITS_HPP_
+
+#include "artemis.hpp"
+
+#include <cmath>
+
+namespace ArtemisUtils {
+
+class Units {
+ public:
+  Units(ParameterInput *pin);
+
+  // Unit conversions
+  Real GetLengthCodeToPhysical() const { return length_; }
```

Comment:
Should these all be accessible on device?

Reply:
Yes; done. Class can only be constructed on device though because it interacts with `ParameterInput`. Device-usable copy constructor provided for getting this into kernels though.